hack/e2e-common.sh

```diff
@@ -0,0 +1,181 @@
+    # ssh-keygen will put the public key here
+    TMP_PUB=${SSH_PRIV_KEY_PATH}.pub
+    # Answer 'y' to the overwrite prompt, since we touched the file
+    yes y | ssh-keygen -q -t rsa -N '' -f ${SSH_PRIV_KEY_PATH}
+    # Now put the pubkey where we expected it
+    mv ${TMP_PUB} ${SSH_PUBLIC_KEY_FILE}
+fi
+
+PULL_SECRET_FILE="${PULL_SECRET_FILE:-${CLUSTER_PROFILE_DIR}/pull-secret}"
+export HIVE_NS="hive-e2e"
+export HIVE_OPERATOR_NS="hive-operator"
+
+# Install Hive
+IMG="${HIVE_IMAGE}" make deploy
+
+function save_hive_logs() {
+  oc logs -n "${HIVE_NS}" deployment/hive-controllers > "${ARTIFACT_DIR}/hive-controllers.log"
+  oc logs -n "${HIVE_NS}" deployment/hiveadmission > "${ARTIFACT_DIR}/hiveadmission.log"
+}
+
+SRC_ROOT=$(git rev-parse --show-toplevel)
+
+USE_MANAGED_DNS=${USE_MANAGED_DNS:-true}
+
+case "${CLOUD}" in
+"aws")
+	CREDS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+        # Accept creds from the env if the file doesn't exist.
+        if ! [[ -f $CREDS_FILE ]] && [[ -n "${AWS_ACCESS_KEY_ID}" ]] && [[ -n "${AWS_SECRET_ACCESS_KEY}" ]]; then
+            # TODO: Refactor contrib/pkg/adm/managedns/enable::generateAWSCredentialsSecret to
+            # use contrib/pkg/utils/aws/aws::GetAWSCreds, which knows how to look for the env
+            # vars if the file isn't specified; and use this condition to generate (or not)
+            # the whole CREDS_FILE_ARG="--creds-file=${CREDS_FILE}".
+            printf '[default]\naws_access_key_id=%s\naws_secret_access_key=%s\n' "$AWS_ACCESS_KEY_ID" "$AWS_SECRET_ACCESS_KEY" > $CREDS_FILE
+        fi
+	BASE_DOMAIN="${BASE_DOMAIN:-hive-ci.openshift.com}"
+	EXTRA_CREATE_CLUSTER_ARGS="--aws-user-tags expirationDate=$(date -d '4 hours' --iso=minutes --utc)"
+	;;
+"azure")
+	CREDS_FILE="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+	BASE_DOMAIN="${BASE_DOMAIN:-ci.azure.devcluster.openshift.com}"
+	;;
+"gcp")
+	CREDS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+	BASE_DOMAIN="${BASE_DOMAIN:-origin-ci-int-gce.dev.openshift.com}"
+	;;
+*)
+	echo "unknown cloud: ${CLOUD}"
+	exit 1
+	;;
+esac
+
+if $USE_MANAGED_DNS; then
+	# Generate a short random shard string for this cluster similar to OSD prod.
+	# This is to prevent name conflicts across customer clusters.
+	CLUSTER_SHARD=$(cat /dev/urandom | tr -dc 'a-z' | fold -w 8 | head -n 1)
+	CLUSTER_DOMAIN="${CLUSTER_SHARD}.${BASE_DOMAIN}"
+	go run "${SRC_ROOT}/contrib/cmd/hiveutil/main.go" adm manage-dns enable ${BASE_DOMAIN} \
+		--creds-file="${CREDS_FILE}" --cloud="${CLOUD}"
+	MANAGED_DNS_ARG=" --manage-dns"
+else
+	CLUSTER_DOMAIN="${BASE_DOMAIN}"
+fi
+
+
+echo "Using cluster base domain: ${CLUSTER_DOMAIN}"
+
+function capture_manifests() {
+    oc get clusterdeployment -A -o yaml &> "${ARTIFACT_DIR}/hive_clusterdeployment.yaml" || true
+    oc get clusterimageset -o yaml &> "${ARTIFACT_DIR}/hive_clusterimagesets.yaml" || true
+    oc get clusterprovision -A -o yaml &> "${ARTIFACT_DIR}/hive_clusterprovision.yaml" || true
```

Comment:
could you add clusterstate -A -o yaml here as well, may give us some insight into operators in the cluster.

Reply:
Done.